src/containers/SearchPage/SearchPage.shared.js

```diff
@@ -45,23 +45,30 @@ const validURLParamForListingTypeData = (listingTypes, param) => {
  * @param {Object} filterConfigs contains listingFieldsConfig and defaultFiltersConfig.
  * @returns search parameters without currently restricted listing fields
  */
-export const omitLimitedListingFieldParams = (searchParams, filterConfigs) => {
+export const omitLimitedListingFieldParams = (searchParams, filterConfigs, pathParams = {}) => {
   const {
     listingFieldsConfig,
     defaultFiltersConfig,
     listingCategories,
     activeListingTypes,
-    listingTypePathParam,
   } = filterConfigs;
+  const { listingTypePathParam } = pathParams;
   const categorySearchConfig = defaultFiltersConfig.find(f => f.schemaType === 'category');
   const listingTypeSearchConfig = defaultFiltersConfig.find(f => f.schemaType === 'listingType');
   const validNestedCategoryParamNames = categorySearchConfig
     ? validURLParamForCategoryData(categorySearchConfig.key, listingCategories, 1, searchParams)
     : {};
 
-  const validListingTypeParamNames = listingTypeSearchConfig
-    ? validURLParamForListingTypeData(activeListingTypes, searchParams)
+  const listingTypeParamMaybe = listingTypePathParam
+    ? { pub_listingType: listingTypePathParam }
     : {};
+  const validListingTypeParamNames =
+    activeListingTypes && listingTypeSearchConfig
+      ? validURLParamForListingTypeData(activeListingTypes, {
+          ...searchParams,
+          ...listingTypeParamMaybe,
+        })
+      : {};
```

Comment:
I think you should do this earlier!

E.g. in SearchPageWithMap:
```js
        return {
          currentQueryParams: omitLimitedListingFieldParams(
            {
              ...mergedQueryParams,
              ...updatedURLParams,
              ...(listingTypePathParam ? { pub_listingType: listingTypePathParam } : {}),
              ...keywordsMaybe,
              ...datesAndSeatsMaybe,
              address,
              bounds,
            },
            filterConfigs,
            currentPathParams
          ),
```

Reply:
I.e. omitLimitedListingFieldParams would keep the original parameter signature - and pub_listingType is enforced to be the one that's found from the path parameters.

Reply:
I tested this now, and with this change the pub_listingType param gets passed to the URL even when the listingTypePathParam is defined i.e. when it shouldn't be included. 

So it turns out the omitLimitedListingFieldParams actually does need the information on whether this is a search param or a path param – it returns the search params that get set to the URL, and we don't want to set pub_listingType to URL in all cases.

Reply:
Had a conversation on this IRL, and decided to pass currentPathParams in filterConfig to these functions